dspace-server-webapp/src/main/java/org/dspace/app/rest/submit/step/DescribeStep.java

```diff
@@ -110,13 +113,56 @@ private void readField(InProgressSubmission obj, SubmissionStepConfig config, Da
     }
 
     @Override
-    public void doPatchProcessing(Context context, Request currentRequest, InProgressSubmission source, Operation op)
-        throws Exception {
-
-        PatchOperation<MetadataValueRest> patchOperation = new PatchOperationFactory()
-            .instanceOf(DESCRIBE_STEP_METADATA_OPERATION_ENTRY, op.getOp());
-        patchOperation.perform(context, currentRequest, source, op);
-
+    public void doPatchProcessing(Context context, Request currentRequest, InProgressSubmission source, Operation op,
+                                  SubmissionStepConfig stepConf) throws Exception {
+
+        String[] pathParts = op.getPath().substring(1).split("/");
+        DCInputSet inputConfig = inputReader.getInputsByFormName(stepConf.getId());
+        if ("remove".equals(op.getOp()) && pathParts.length < 3) {
+            // manage delete all step fields
+            String[] path = op.getPath().substring(1).split("/", 3);
+            String configId = path[1];
+            List<String> fieldsName = getInputFieldsName(inputConfig, configId);
+            for (String fieldName : fieldsName) {
+                String fieldPath = op.getPath() + "/" + fieldName;
+                Operation fieldRemoveOp = new RemoveOperation(fieldPath);
+                PatchOperation<MetadataValueRest> patchOperation = new PatchOperationFactory()
+                     .instanceOf(DESCRIBE_STEP_METADATA_OPERATION_ENTRY, fieldRemoveOp.getOp());
+                patchOperation.perform(context, currentRequest, source, fieldRemoveOp);
+            }
+        } else {
+            PatchOperation<MetadataValueRest> patchOperation = new PatchOperationFactory()
+                        .instanceOf(DESCRIBE_STEP_METADATA_OPERATION_ENTRY, op.getOp());
+            String[] split = patchOperation.getAbsolutePath(op.getPath()).split("/");
+            if (inputConfig.isFieldPresent(split[0])) {
+                patchOperation.perform(context, currentRequest, source, op);
+            } else {
+                throw new UnprocessableEntityException("The field " + split[0] + " is not present in section "
```

Comment:
this is not verified for remove. Does this imply the user can e.g. remove the item template, even if that metadata field is not adjustable in the form?
Can you add an IT to prove the user can't remove a field not in the form?

Reply:
@abollini can you also verify this feedback?

Reply:
@benbosman I guess that this is the test that you have requested https://github.com/DSpace/DSpace/pull/2760/commits/4ff01ae963a4bb765b3464600c005b4aa51d15c0#diff-38ba2da3dd74a1027593bbd62d684626R3792
the dc.title is not present in the traditionalpagetwo form so the remove patch fails and the dc.title is still here

Reply:
Thanks, that's indeed the test I was looking for